### PR TITLE
Implement Feature #3 - Implement UpDown Game Logic

### LINKED ISSUE #3: Issue #3 - Implement UpDown Game Logic
## Summary
Develop the "Number Up-Down" mini-game where the user guesses a number between 1 and 100.
The game should provide feedback ("Up" or "Down") and integrate with 'UserManager' to save the best score upon completion.

## To-Do List
- [ ] **Game Loop Implementation**:
	- Generate a random number (1-100)
	- Receive user input via 'Scanner'.
	- Provide "Up", "Down", or "Correct" feedback.
- [ ] **Rule Logic**
	- Limit attempts to 10.
	- Calculate score based on remaining attempts.
- [ ] **Integration**
	- Update the current user's score if the new score is larger.
	- Ensure the game exits gracefully to the main menu.

## Class Structure
- **UpDownGame (implements 'MiniGame')**

## Related Issue
- **2 (Implement UserManager)**: The game requires 'UserManager' to verify and save the high score after the game ends.

## Acceptance Criteria
- [ ] Must correctly identify number between 1 and 100.
- [ ] Must show "Game Over" after 10 failed attempts.
- [ ] Integration test with 'UserManager' passes. (Score update)

## Branch
- 'feature/game-updown'

### PR DESCRIPTION
**Pull request by ichbinju0**
## Linked Issue
Closes #3 

## Description
- Implemented the core Number UpDown game logic (1-100 guessing with a 10-attempt limit).
- Added score calculation based on remaining attempts (10 - attempts + 1).
- Integrated UX improvement allowing the user to quit mid-game by inputting '0'.
- Added comprehensive unit tests (UpDownTest.java) to verify game rules and boundary conditions.

## Commit Type
- [x] `[FEAT]` New feature
- [ ] `[FIX]` Bug fix 
- [ ] `[REFACTOR]` Code restructuring 
- [ ] `[CHORE]` Build tasks, config changes
- [x] `[TEST]` Adding or updating tests 
- [ ] `[DOCS]` Documentation changes 

## Testing Validation
- [x] Ran `mvn test` and verified that the build and tests pass.

## PR Checklist
- [x] Added Reviewers (at least 1 team member to review).
- [x] Added Assignees (assigning myself or others doing this task).

### CODE REVIEW

**Review by so-yeon-333 — CHANGES_REQUESTED** [review]
nextLine is much safer!

File: src/main/java/minimanimo/game/UpDown.java [hunk 1]
```diff
@@ -9,14 +10,56 @@ public String getGameName() {
...
+        int targetNumber = random.nextInt(100) + 1;
+        int attempts = 0;
+        int maxAttempts = 10;
+
+        System.out.println("Welcome to UpDown! Guess a number between 1 and 100.");
+        System.out.println("You have " + maxAttempts + " attempts to guess the correct number.");
+
+        while (attempts < maxAttempts) {
+            System.out.println("Attempt: " + (attempts + 1) + ". Enter your guess: "); // Prompt user for input
+            String input = scanner.next().trim();
```
[the remark this answers is not in the document]
Comment (so-yeon-333) [reply src/main/java/minimanimo/game/UpDown.java]: The leftover newline character causes the menu input to be skipped when returning to the main menu. I hope no bugs in our game! Good luck!

**Review by bbirribbarribbo — COMMENTED** [review]
Everything looks good, except I should change .next() to .nextLine() for safety. LGTM!!

**Review by so-yeon-333 — CHANGES_REQUESTED** [review]
Check this. I think 0 should not be inside here.

(add more info: in the test  testInvalidInputAndGameOver() )

**Review by so-yeon-333 — CHANGES_REQUESTED** [review]
Great job, just requiring a minor fix.

**Review by Gyeongyoon — CHANGES_REQUESTED** [review]
Great job on implementing the logic! Just fix the test case order regarding the exit command ("0"), and it looks good to merge. LGTM! 🚀

File: src/test/java/minimanimo/game/UpDownTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        UpDown game = new UpDown();
+        assertFalse(game.isLowerScoreBetter());
+    }
+
+    @Test
+    void testInvalidInputAndGameOver() {
+        String input = "abc\n" +
+                "def\n" +
+                " \n" +
+                "0\n" +
```
[the remark this answers is not in the document]
Comment (ichbinju0) [reply src/test/java/minimanimo/game/UpDownTest.java]: I fix this. Thank you for the feedback. Please check.

**Comment by ichbinju0** [comment]
> > > Check this. I think 0 should not be inside here.
> > 
> > 
> > Can you please tell me exactly where you mean?
> 
> Oh I'm sorry.
> 
> ```
> @Test
>     void testInvalidInputAndGameOver() {
>         String input = "abc\n" +
>                 "def\n" +
>                 " \n" +
>                 "0\n" + <- here.
>                 "101\n" +
>                 "-50\n" +
>                 "1\n2\n3\n4\n5\n6\n7\n8\n9\n10\n";
> 
>         ByteArrayInputStream in = new ByteArrayInputStream(input.getBytes());
>         Scanner scanner = new Scanner(in);
>         User user = new User("TestUser");
>         UpDown game = new UpDown();
> 
>         int score = game.startGame(user, scanner);
> 
>         assertTrue(score >= 0);
>     } 
> ```

Thank you for the feedback. I couldn't check the test file because I was adding a function in the middle. I fixed it, so please check it.!!

**Review by so-yeon-333 — APPROVED** [review]
Thanks for changing all the things what I said earlier. You did a great job!

**Review by Gyeongyoon — APPROVED** [review]
LGTM!
